datafusion/physical-plan/src/joins/nested_loop_join.rs

```diff
@@ -689,6 +674,8 @@ enum NestedLoopJoinStreamState {
     ProcessProbeBatch(RecordBatch),
     /// Indicates that probe-side has been fully processed
     ExhaustedProbeSide,
+    /// Output unmatched build-side rows
```

Comment:
```suggestion
    /// Output unmatched build-side rows.
    /// The indices for rows to output has already been calculated in the previous
    /// `ExhaustedProbeSide` state. In this state the final batch will be materialized
    // incrementally.
    // The inner `RecordBatch` is an empty dummy batch used to get right schema.
```
Maybe we can also rename `ExhaustedProbeSide` to `PrepareUnmatchedBuildRows` to be more accurate.